### DISCUSSION TITLE
Problem with test a json page

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#44314] [Q&A] [answered]
I have a `Json` page that has a key called `firstName` and I created a test for it as follows:
```
<?php

namespace App\Tests\Controller;

use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;

class JsonControllerTest extends WebTestCase
{
    public function testJson(): void
    {
        $client = static::createClient();
        $crawler = $client->request('POST', '/json-page');
        $response = $client->getResponse()->getContent();
        $this->assertResponseIsSuccessful();
        $this->assertArrayHasKey('firstName', (array)$response);
    }
}
```
Now after starting the test, this error appears:
`Failed asserting that an array has the key 'firstName'.`
How can I fix this problem? Is my code wrong?

**Comment by BahmanMehrdad** [comment, marked as answer]
`assertArrayHasKey` not worked because `$response` type is string and when you convert it to array, the key of that is `0`.
So you can use `assertStringContainsString` like this:

`$this->assertStringContainsString('firstName', $response);`

**Reply by karaka200** [reply]
Thank you

**Comment by GromNaN** [comment]
For such tests, I added a method to decode JSON (in a trait).

```php
trait JsonResponse
{
    protected function getResponseContent(): array
    {
        return json_decode($this->client->getResponse()->getContent(), true);
    }
}
```

**Comment by stloc** [comment]


**Reply by stof** [reply]
please start a dedicated discussion providing a full reproducing case instead of commenting on a conversation marked as resolved 3 years ago.

**Reply by stloc** [reply]
Hi,
sorry this was the only discussion with the same error, I thought I was doing the right thing
